src/redux/actions/planActions.js

```diff
@@ -0,0 +1,329 @@
+    isSynced: false
+  };
+}
+
+function getPlansFailure() {
+  return {
+    type: GET_PLANS_FAILURE,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: false
+  };
+}
+
+function getPlansSuccess(plans) {
+  return {
+    type: GET_PLANS_SUCCESS,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: true,
+    plans: plans
+  };
+}
+
+function addPlanRequest() {
+  return {
+    type: ADD_PLAN_REQUEST,
+    lastUpdated: Date.now(),
+    isFetching: true,
+    isSynced: false
+  };
+}
+
+function addPlanFailure() {
+  return {
+    type: ADD_PLAN_FAILURE,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: false
+  };
+}
+
+function addPlanSuccess() {
+  return {
+    type: ADD_PLAN_SUCCESS,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: true
+  };
+}
+
+function updatePlanRequest() {
+  return {
+    type: UPDATE_PLAN_REQUEST,
+    lastUpdated: Date.now(),
+    isFetching: true,
+    isSynced: false
+  };
+}
+
+function updatePlanFailure() {
+  return {
+    type: UPDATE_PLAN_FAILURE,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: false
+  };
+}
+
+function updatePlanSuccess(plans) {
+  return {
+    type: UPDATE_PLAN_SUCCESS,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: true,
+    plans: plans
+  };
+}
+
+function deletePlanRequest() {
+  return {
+    type: DELETE_PLAN_REQUEST,
+    lastUpdated: Date.now(),
+    isFetching: true,
+    isSynced: false
+  };
+}
+
+function deletePlanFailure() {
+  return {
+    type: DELETE_PLAN_FAILURE,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: false
+  };
+}
+
+function deletePlanSuccess(plans) {
+  return {
+    type: DELETE_PLAN_SUCCESS,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: true,
+    plans: plans
+  };
+}
+
+function getPlanInfoRequest() {
+  return {
+    type: GET_PLAN_INFO_REQUEST,
+    lastUpdated: Date.now(),
+    isFetching: true,
+    isSynced: false
+  };
+}
+
+function getPlanInfoFailure() {
+  return {
+    type: GET_PLAN_INFO_FAILURE,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: false
+  };
+}
+
+function getPlanInfoSuccess(plans) {
+  return {
+    type: GET_PLAN_INFO_SUCCESS,
+    lastUpdated: Date.now(),
+    isFetching: false,
+    isSynced: true,
+    plans: plans
+  };
+}
+
+/**************************** Logical functions ****************************/
+
+/**
+ * Performs an http POST get plan info request to server.
+ * Dispatches getUserInfoRequest to indicate the beginning of a getPlanInfo process.
+ * Dispatches getPlanInfoFailure to indicate the end of a failed getPlanInfo process.
+ * Dispatches getPlanSuccess to indicate the end of a successful getPlanInfo process.
+ * If getPlanInfo process succeeds, a plan list object is received and passed into
+ * getPlanInfoSuccess to be stored into state.
+ * @returns {function(*): Promise<Response | never>} dispatch results.
```

Comment:
Add the @param e_id comment to clarify that e_id is the event id for the plan

Reply:
Can do.